ConwaysGameOfLife/GameController.cs

```diff
@@ -76,51 +84,61 @@ public void Run()
                         _output.WriteLine($"or {Messages.Play}");
 
                         input = _input.ReadLine();
-
-                        // if (UserEndsGame(input))
+            
+                        if (UserEndsGame(input))
+                        {
+                            _output.WriteLine(Messages.GameEnd);
+                            return;
+                        }
+                        // if (UserPressesPToPlay(input))
                         // {
-                        //     _output.WriteLine(Messages.GameEnd);
                         //     return;
                         // }
-
                         var isValidLocation = Validator.IsValidLocation(input, _universe.GridWidth, _universe.GridLength);
                         while (!isValidLocation)
                         {
                             _output.WriteLine(Messages.InvalidInput);
                             _output.WriteLine($"{Messages.RequestLiveCell}.");
                             input = _input.ReadLine();
 
-                            // if (UserEndsGame(input))
+                            if (UserEndsGame(input))
+                            {
+                                _output.WriteLine(Messages.GameEnd);
+                                return;
+                            }
+
+                            // if (UserPressesPToPlay(input))
                             // {
-                            //     _output.WriteLine(Messages.GameEnd);
                             //     return;
                             // }
+                            
                             isValidLocation = Validator.IsValidLocation(input, _universe.GridWidth, _universe.GridLength);
                         }
                         var location = InputParser.ParseLocation(input);
 
                         SetLiveCellLocation(location);
                         DisplayUniverse();
                     }
-                    while (input != "p");
+                    while (!UserPressesPToPlay(input));

```

Comment:
```suggestion
                    while (!UserPressesPToPlay(input));
```
another potential infinite loop here as you're never updating `input` i.e. if you get into this loop because `input` is not `"p"`, you'll never get out